### PR TITLE
Layout v3 continued

### PR DESCRIPTION
**Pull request by knsv**
## :bookmark_tabs: Summary

Adding elkjs as another renderer for flowcharts.

## :straight_ruler: Design Decisions

Describe the way your implementation works or what design decisions you made if applicable.

### :clipboard: Tasks

Make sure you

- [x] :book: have read the [contribution guidelines](https://github.com/mermaid-js/mermaid/blob/develop/CONTRIBUTING.md)
- [ ] :computer: have added unit/e2e tests (if appropriate)
- [x] :notebook: have added documentation (if appropriate)
- [x] :bookmark: targeted `develop` branch

### CODE REVIEW

File: packages/mermaid/src/docs/syntax/flowchart.md [hunk 1]
```diff
@@ -669,7 +669,21 @@ flowchart LR
     C -->|Two| E[Result two]
 ```
 
-## Configuration...
+## Configuration
+
+### Renderer
+
+The layout of the flowchart diagram can be don in multiple ways. The default rendering is done using an algorithm called dagre. It is possible to use another algorithm called **elk** for rendering flowcharts which is better for large and more complex diagrams. This is an expermental feature but worth exploring.
```
Comment (weedySeaDragon) [inline packages/mermaid/src/docs/syntax/flowchart.md:676]: _Suggested changes:_


The layout of the diagram is done with the _renderer._  The default renderer is **dagre.** 

Starting with Mermaid version 9.4, you can use an alternate renderer named **elk**.   The _elk_ renderer is better for larger and/or more complex diagrams. 
```note
The _elk_ renderer is an experimenal feature.
```

You can change the renderer to _elk_ by adding this directive:

File: packages/mermaid/src/docs/syntax/flowchart.md [hunk 2]
```diff
@@ -669,7 +669,21 @@ flowchart LR
...
+
+This can be enabled using directives by adding this in the diagram code:
+
+```
+%%{init: {"flowchart": {"defaultRenderer": "elk"}} }%%
+```
+
+Note that the site needs to use mermaid version 9.4+ for this to work and have this featured enabled in the lazy-loading configuration.
```
Comment (weedySeaDragon) [inline packages/mermaid/src/docs/syntax/flowchart.md:684]: incorporate this in the first paragraph. See my suggested change above

**Review by weedySeaDragon — COMMENTED** [review]
I have some suggestions for the documentation.  (I didn't review the other changes.)

**Comment by sidharthv96** [comment]
@knsv is this ready for review or WIP?

File: packages/mermaid/src/docs/syntax/flowchart.md (hunk 2 above)
Comment (knsv) [reply packages/mermaid/src/docs/syntax/flowchart.md:684]: will do!

File: packages/mermaid/src/diagrams/flowchart/parser/flow.jison [hunk 3]
```diff
@@ -82,6 +82,7 @@ that id.
 <click>[\s\n]           this.popState();
 <click>[^\s\n]*         return 'CLICK';
 
+"cyto"                 {if(yy.lex.firstGraph()){this.begin("dir");}  return 'GRAPH';}
```
Comment (ashishjain0512) [inline packages/mermaid/src/diagrams/flowchart/parser/flow.jison:85]: Do we really need this?

Comment (knsv) [reply packages/mermaid/src/diagrams/flowchart/parser/flow.jison]: No, you are right. Fixing it!

**Review by ashishjain0512 — APPROVED** [review]
This is really good stuff. Will be super helpful for complex flow diagrams

**Pull request merged**